Describe Temurin preference of the Jenkins project (#6102)
jenkins-infra/jenkins.io: content/doc/administration/requirements/java.adoc

and notify users when they are launched with an unsupported version.

The plugin:versioncolumn[Versions Node Monitors plugin] provides detailed Java version monitoring.

> ##Java Development Kits Used
> 
> Jenkins uses Eclipse Temurin as its primary JDK for building and testing Java based applications. This includes:
> * Container Images
> * Controller Builds
> * Testing Infrastructure
> 
> Some of the reasons of choosing Temurin are:
> 1. Its availability over a wide range of platforms, including different operating systems and architectures, over many different Java SE versions.
> 2. Regular maintainence and long term support provided by the Eclipse Foundation
> 3. Additional performance optimizations designed to improve the performance and scalability of the JDK in enterprise-level applications.

Reviewer: ```suggestion
== Java Development Kits Used

Jenkins uses Eclipse Temurin as its primary JDK for building and testing Java based applications.
This includes:
* Container Images
* Controller Builds
* Testing Infrastructure

Some of the reasons for choosing Temurin are:
. Its availability over a wide range of platforms, including different operating systems and architectures, over many different Java SE versions.
. Regular maintenance and long term support provided by the Eclipse Foundation.
. Additional performance optimizations designed to improve the performance and scalability of the JDK in enterprise-level applications.
```
updated to use ascii formatting (we have moved from markdown to asciidoc formatting) and set the ordered list accordingly.

and notify users when they are launched with an unsupported version.

The plugin:versioncolumn[Versions Node Monitors plugin] provides detailed Java version monitoring.

== Java Development Kits Used

Jenkins uses Eclipse Temurin as its primary JDK for building and testing Java based applications.

> This includes:
> * Container Images
> * Controller Builds
> * Testing Infrastructure
> 

Reviewer 2: ```suggestion
This includes:

* Container Images
* Controller Builds
* Testing Infrastructure

```

> Some of the reasons for choosing Temurin are:
> . Its availability over a wide range of platforms, including different operating systems and architectures, over many different Java SE versions.
> . Regular maintenance and long term support provided by the Eclipse Foundation.
> . Additional performance optimizations designed to improve the performance and scalability of the JDK in enterprise-level applications.
> 

Reviewer 2: ```suggestion
Some of the reasons for choosing Temurin are:

* It's availability over a wide range of platforms, including different operating systems and architectures, over many different Java SE versions.
* Regular maintenance and long term support provided by the Eclipse Foundation.
* Additional performance optimizations designed to improve the performance and scalability of the JDK in enterprise-level applications.
```

and notify users when they are launched with an unsupported version.

The plugin:versioncolumn[Versions Node Monitors plugin] provides detailed Java version monitoring.

> == Java Development Kits Used

Reviewer 3: Since the other entries in the file use `##` to identify headings, this should be the same
```suggestion
## Java Development Kits Used
```

> Jenkins uses Eclipse Temurin as its primary JDK for building and testing Java based applications.

Reviewer 3: Optional - hyperlink to the Eclipse Temurin project in case readers want to know more about it.
```suggestion
Jenkins uses link:https://projects.eclipse.org/projects/adoptium.temurin[Eclipse Temurin] as its primary JDK for building and testing Java based applications.
```

> Jenkins uses Eclipse Temurin as its primary JDK for building and testing Java based applications.

Reviewer 4: It may be confusing that in this senteence "Jenkins" refers to the organization and not to the software.

Reviewer 3: Agreed that is a good idea.
```suggestion
The Jenkins project uses Eclipse Temurin as its primary JDK for building and testing Java based applications.
```

This includes:

> * Container Images

Reviewer 4: I think it would be worth explaining what "controller builds" are in this context.

> * Controller Builds

Reviewer 3: Optional - we also use Temurin for automated plugin releases.
```suggestion
* Controller Builds
* link:/doc/developer/publishing/releasing-cd/[Automated plugin releases]
```

* Testing Infrastructure

Some of the reasons for choosing Temurin are:

> * It's availability over a wide range of platforms, including different operating systems and architectures, over many different Java SE versions.

Reviewer: ```suggestion
* Availability over a wide range of platforms, including different operating systems and architectures, over many different Java SE versions.
```
Otherwise, it would be `Its availability over a wide range of platforms, including different operating systems and architectures, over many different Java SE versions.`

* Regular maintenance and long term support provided by the Eclipse Foundation.

> * Additional performance optimizations designed to improve the performance and scalability of the JDK in enterprise-level applications.

Reviewer 3: I could not find any Temurin page that claimed they were adding additional performance optimizations.  Can you point to the location where they make that claim?
```suggestion
```

Author: I'm sorry, I took some help from ChatGPT. I  guess I should have been more careful

Reviewer 3: That's impressive.  The text is well phrased and the claim is believable that it might come from an organization that creates JDK distributions.  If you can find an attribution for that text, I'm happy to see it included.  Otherwise, we leave it with the first two items

Reviewer 5: not really impressive, it's one of the biggest pitfalls of it, confidence and believable but no references and often wrong

and notify users when they are launched with an unsupported version.

The plugin:versioncolumn[Versions Node Monitors plugin] provides detailed Java version monitoring.

> ## Java Development Kits Used

Reviewer 3: ```suggestion
## Java Development Kit
```
